### PR TITLE
JAMES-2855 diverses testing libs upgrade

### PR DESCRIPTION
**Pull request by Arsnael**
Note : should remove com.jaway.awaitility for only keeping org.awaitility

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -2543,7 +2543,7 @@
             <dependency>
                 <groupId>org.awaitility</groupId>
```
Comment (chibenwa) [inline pom.xml]: Shouldn't we be merging com.jayway.awaitility:awaitility & org.awaitility:awaitility ?

Comment (Arsnael) [reply pom.xml]: WIP :)

**Comment by chibenwa** [comment]
```
[996a7820043e53f6e76c40890c9715ad4d7ddaf8] [ERROR]   MappingSourceTest.shouldRespectBeanContract:41 EqualsVerifier found a problem in class MappingSource.
[996a7820043e53f6e76c40890c9715ad4d7ddaf8] -> Significant fields: equals should not use wildcard, but it does.
```

File: mailbox/api/src/test/java/org/apache/james/mailbox/events/ErrorHandlingContract.java [hunk 2]
```diff
@@ -297,7 +291,19 @@ default void redeliverShouldNotSendEventsToKeyListeners() {
         eventBus().reDeliver(GROUP_A, EVENT).block();
 
         WAIT_CONDITION
-            .until(() -> assertThat(eventCollector.getEvents()).hasSize(1));
+            .until(() -> eventCollector.getEvents().size() == 1);
         assertThat(eventCollector2.getEvents()).isEmpty();
     }
+
+    default boolean deadLetterContainsOnlyAnEvent() {
```
Comment (Arsnael) [inline mailbox/api/src/test/java/org/apache/james/mailbox/events/ErrorHandlingContract.java]: Not proud at all about this method... I feel we can probably do better here to check that but... I couldn't. If someone wants to give it a try... :)

**Comment by Arsnael** [comment]
```
[bd0f6b4c05450de7dfe88615623c1c7162731d25] [ERROR] Failures: 
[bd0f6b4c05450de7dfe88615623c1c7162731d25] [ERROR]   SieveQuotaRoutesTest.getGlobalSieveQuotaShouldReturn204WhenNoQuotaSet:67 1 expectation failed.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] Expected status code <204> but was <404>.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] 
[bd0f6b4c05450de7dfe88615623c1c7162731d25] [ERROR]   SieveQuotaRoutesTest.getGlobalSieveQuotaShouldReturnStoredValue:79 1 expectation failed.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] Expected status code <200> but was <404>.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] 
[bd0f6b4c05450de7dfe88615623c1c7162731d25] [ERROR]   SieveQuotaRoutesTest.getPerUserQuotaShouldReturn204WhenNoQuotaSetForUser:142 1 expectation failed.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] Expected status code <204> but was <404>.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] 
[bd0f6b4c05450de7dfe88615623c1c7162731d25] [ERROR]   SieveQuotaRoutesTest.getPerUserSieveQuotaShouldReturnStoredValue:154 1 expectation failed.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] Expected status code <200> but was <404>.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] 
[bd0f6b4c05450de7dfe88615623c1c7162731d25] [ERROR]   SieveQuotaRoutesTest.removeGlobalSieveQuotaShouldRemoveGlobalSieveQuota:134 1 expectation failed.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] Expected status code <204> but was <404>.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] 
[bd0f6b4c05450de7dfe88615623c1c7162731d25] [ERROR]   SieveQuotaRoutesTest.removeGlobalSieveQuotaShouldReturn204WhenNoQuotaSet:124 1 expectation failed.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] Expected status code <204> but was <404>.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] 
[bd0f6b4c05450de7dfe88615623c1c7162731d25] [ERROR]   SieveQuotaRoutesTest.removePerUserSieveQuotaShouldRemoveQuotaForUser:209 1 expectation failed.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] Expected status code <204> but was <404>.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] 
[bd0f6b4c05450de7dfe88615623c1c7162731d25] [ERROR]   SieveQuotaRoutesTest.removePerUserSieveQuotaShouldReturn204WhenNoQuotaSetForUser:199 1 expectation failed.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] Expected status code <204> but was <404>.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] 
[bd0f6b4c05450de7dfe88615623c1c7162731d25] [ERROR]   SieveQuotaRoutesTest.updateGlobalSieveQuotaShouldReturn400WhenMalformedJSON:107 1 expectation failed.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] Expected status code <400> but was <404>.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] 
[bd0f6b4c05450de7dfe88615623c1c7162731d25] [ERROR]   SieveQuotaRoutesTest.updateGlobalSieveQuotaShouldReturn400WhenRequestedSizeNotPositiveInteger:116 1 expectation failed.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] Expected status code <400> but was <404>.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] 
[bd0f6b4c05450de7dfe88615623c1c7162731d25] [ERROR]   SieveQuotaRoutesTest.updateGlobalSieveQuotaShouldUpdateStoredValue:96 1 expectation failed.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] Expected status code <204> but was <404>.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] 
[bd0f6b4c05450de7dfe88615623c1c7162731d25] [ERROR]   SieveQuotaRoutesTest.updatePerUserSieveQuotaShouldReturn400WhenMalformedJSON:182 1 expectation failed.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] Expected status code <400> but was <404>.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] 
[bd0f6b4c05450de7dfe88615623c1c7162731d25] [ERROR]   SieveQuotaRoutesTest.updatePerUserSieveQuotaShouldReturn400WhenRequestedSizeNotPositiveInteger:191 1 expectation failed.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] Expected status code <400> but was <404>.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] 
[bd0f6b4c05450de7dfe88615623c1c7162731d25] [ERROR]   SieveQuotaRoutesTest.updatePerUserSieveQuotaShouldUpdateStoredValue:171 1 expectation failed.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] Expected status code <204> but was <404>.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] 
[bd0f6b4c05450de7dfe88615623c1c7162731d25] [ERROR]   SieveScriptRoutesTest.defineAddActiveSieveScriptAddScriptSucceededOneWhenNotAddActivateParam:194 1 expectation failed.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] Expected status code <204> but was <404>.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] 
[bd0f6b4c05450de7dfe88615623c1c7162731d25] [ERROR]   SieveScriptRoutesTest.defineAddActiveSieveScriptGetActiveShouldThrowsExceptionWhenAddActivateParamFalse:228 1 expectation failed.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] Expected status code <204> but was <404>.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] 
[bd0f6b4c05450de7dfe88615623c1c7162731d25] [ERROR]   SieveScriptRoutesTest.defineAddActiveSieveScriptInvokeShouldReturnBadRequestWhenAddActivateParamWithNotBooleanValue:244 1 expectation failed.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] Expected status code <400> but was <404>.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] 
[bd0f6b4c05450de7dfe88615623c1c7162731d25] [ERROR]   SieveScriptRoutesTest.defineAddActiveSieveScriptSetActiveTrueWhenAddActivateParamTrue:211 1 expectation failed.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] Expected status code <204> but was <404>.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] 
[bd0f6b4c05450de7dfe88615623c1c7162731d25] [ERROR]   SieveScriptRoutesTest.defineAddActiveSieveScriptShouldReturnBadRequestWhenScriptIsNotSet:166 1 expectation failed.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] Expected status code <400> but was <404>.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] 
[bd0f6b4c05450de7dfe88615623c1c7162731d25] [ERROR]   SieveScriptRoutesTest.defineAddActiveSieveScriptShouldReturnNotFoundWhenScriptNameIsWhiteSpace:129 1 expectation failed.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] Expected status code <400> but was <404>.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] 
[bd0f6b4c05450de7dfe88615623c1c7162731d25] [ERROR]   SieveScriptRoutesTest.defineAddActiveSieveScriptShouldReturnNotFoundWhenUserNameWhiteSpace:151 1 expectation failed.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] Expected status code <400> but was <404>.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] 
[bd0f6b4c05450de7dfe88615623c1c7162731d25] [ERROR]   SieveScriptRoutesTest.defineAddActiveSieveScriptShouldReturnSucceededWhenScriptIsWhiteSpace:178 1 expectation failed.
[bd0f6b4c05450de7dfe88615623c1c7162731d25] Expected status code <204> but was <404>.
```
Probably something missing somewhere... checking

**Comment by Arsnael** [comment]
Issue with rest-assured upgrade... I read a couple of breaking changes in the changelogs so my guess is it's gonna be quite tricky, so I will isolate that upgrade in its own PR

File: mailbox/api/src/test/java/org/apache/james/mailbox/events/ErrorHandlingContract.java [hunk 3]
```diff
@@ -297,7 +291,19 @@ default void redeliverShouldNotSendEventsToKeyListeners() {
         eventBus().reDeliver(GROUP_A, EVENT).block();
 
         WAIT_CONDITION
-            .until(() -> assertThat(eventCollector.getEvents()).hasSize(1));
+            .until(() -> eventCollector.getEvents().size() == 1);
```
Comment (trantienduchn) [inline mailbox/api/src/test/java/org/apache/james/mailbox/events/ErrorHandlingContract.java]: I've just checked, you can replace `.until()` by `untilAsserted()` and all of the assertion lambdas will be intact.

Comment (Arsnael) [reply mailbox/api/src/test/java/org/apache/james/mailbox/events/ErrorHandlingContract.java]: Could be... but if you check what we do with `org.awaitility` in the rest of the code, it seems that we do this way. I prefer keeping things consistent in our code if possible.

Do you agree with that?

Comment (Arsnael) [reply mailbox/api/src/test/java/org/apache/james/mailbox/events/ErrorHandlingContract.java]: However I agree that it allows me to not use that ugly piece of code I wrote in the `deadLetterContainsOnlyAnEvent` method

Comment (trantienduchn) [reply mailbox/api/src/test/java/org/apache/james/mailbox/events/ErrorHandlingContract.java]: > Do you agree with that?

yep

**Pull request closed without merging**